### PR TITLE
Symposium of small improvements to m3 app

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

In anticipation of a design audit, I went through the whole sample, focusing on components. This PR is a collection of small improvements that would be too small for their own PRs. General goals were consistency, alignment with m3.material.io, and some aesthetic polish. There are a few framework issues for which I've added TODOs and will open corresponding bugs. Adding missing components (e.g. app bar, divider) is out of scope. Of course, I'm open to suggestions.

Screenshot 2023-01-17 at 13 35 15

### Components
- add vertical spacing between grouped components
- rebalance lists in 2-list layout
- use same convention for naming, buttons to show components, and usage info
- add tooltips to components when useful
- remove toasts when tapping on things
- display components in rows, with disabled components at the bottom
- simplify buttons and chips
- use ListTile variants for checkboxes and radio buttons
- limit snackbar and bottom sheet width, simplify both

### General
- add support for system brightness. To do this, I had to refactor a few functions into widgets, and add a `darkTheme`
- inversed icons for options so they show the what will happen rather than the current state
- show which seed color is selected

### Elevation
- hide opacity % for shadow-only elevation demo

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by QuncCccccc — APPROVED** [review]
LGTM. Thanks for making the improvements:) I like the system theme change!

**Comment by domesticmouse** [comment]
@QuncCccccc please feel free to land when you are happy this change is complete.

File: experimental/material_3_demo/lib/main.dart [hunk 1]
```diff
@@ -3,14 +3,18 @@
...
 import 'component_screen.dart';
 import 'elevation_screen.dart';
 import 'typography_screen.dart';
 
 void main() {
   runApp(const MaterialApp(
-      debugShowCheckedModeBanner: false, home: Material3Demo()));
+    debugShowCheckedModeBanner: false,
+    home: Material3Demo(),
+  ));
```
Comment (esouthren) [inline experimental/material_3_demo/lib/main.dart:17]: If you've added a trailing comma, I'd add another on L17?

**Review by esouthren — APPROVED** [review]
LGTM

File: experimental/material_3_demo/lib/component_screen.dart [hunk 2]
```diff
@@ -889,26 +862,25 @@ class _ProgressIndicatorsState extends State<ProgressIndicators> {
 
 const List<NavigationDestination> appBarDestinations = [
   NavigationDestination(
-    tooltip: 'Updated component list',
+    tooltip: 'Components',
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply experimental/material_3_demo/lib/component_screen.dart:865]: Indeed, I'll remove the redundant tooltips

File: experimental/material_3_demo/lib/main.dart (hunk 1 above)
Comment (guidezpl) [reply experimental/material_3_demo/lib/main.dart:17]: done!

**Comment by guidezpl** [comment]
There appears to be an unrelated error

```
info - lib/cookbook.dart:11:1 - 'package:webdriver/io.dart' is deprecated and shouldn't be used. Use 'package:webdriver/async_io.dart' instead.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
```

**Comment by domesticmouse** [comment]
Please update with the latest from `main`. It should fix the issues with CI.

**Comment by HansMuller** [comment]
Nearly all of the changes LGTM, in fact I had a list of 3 requests that already you've covered here! The only change I think we should undo is the common buttons card's layout. The rest of the UI has components organized into neat rows and columns so I think it would be preferable to restore the button rows. Currently the herd layout doesn't quite fit and it looks a little out of place:

Screenshot 2023-01-18 at 10 07 58 AM

**Comment by QuncCccccc** [comment]
> Nearly all of the changes LGTM, in fact I had a list of 3 requests that already you've covered here! The only change I think we should undo is the common buttons card's layout. The rest of the UI has components organized into neat rows and columns so I think it would be preferable to restore the button rows. Currently the herd layout doesn't quite fit and it looks a little out of place:
> 
> Screenshot 2023-01-18 at 10 07 58 AM

I just created a PR to show a simple menu when we click the "more" button on the bottom app bar. I can add one more commit to undo the common button part if it's needed:)

**Comment by guidezpl** [comment]
> Nearly all of the changes LGTM, in fact I had a list of 3 requests that already you've covered here! The only change I think we should undo is the common buttons card's layout. The rest of the UI has components organized into neat rows and columns so I think it would be preferable to restore the button rows. Currently the herd layout doesn't quite fit and it looks a little out of place:
> 
> Screenshot 2023-01-18 at 10 07 58 AM

For the buttons, I'm not sure what you mean. They're currently implemented as a `Column` and three `Row`s (similarly to how I did `Chips`). I had to make it horizontally scrollable due to the space taken up by the row with icons. Maybe we could move that row down one to make it look neater?

image

**Pull request merged**